lib/phoenix_live_view/channel.ex

```diff
@@ -308,6 +297,39 @@ defmodule Phoenix.LiveView.Channel do
     end
   end
 
+  defp inner_component_handle_event(socket, component_socket, component, "lv:clear-flash", val) do
+    component_socket =
+      case val do
+        %{"key" => key} -> Utils.clear_flash(component_socket, key)
+        _ -> Utils.clear_flash(component_socket)
+      end
+
+    component_socket = Utils.maybe_call_update!(socket, component, component_socket.assigns)
```

Comment:
I am not sure if we should call `maybe_call_update!` here. We are definitely not passing new parameters to the socket.

Reply:
I'm calling it to force an update, otherwise the flash isn't cleaned up in the template:
```
  1) test lv:clear-flash component (Phoenix.LiveView.FlashIntegrationTest)
     test/phoenix_live_view/integrations/flash_test.exs:198
     Assertion with =~ failed
     code:  assert result =~ "component[]:info"
     left:  "uri[http://localhost:4000/flash-root]\nroot[]:info\nroot[]:error\n<div id=\"flash-component\" data-phx-component=\"0\">\ncomponent[ok!]:info\ncomponent[]:error\n</div>\nchild[<div data-phx-parent-id=\"phx-Ff-hFHPDVVgmY1oC\" data-phx-session=\"SFMyNTY.g3QAAAACZAAEZGF0YWgCYQR0AAAAB2QAAmlkbQAAAAtmbGFzaC1jaGlsZGQACnBhcmVudF9waWRnZAANbm9ub2RlQG5vaG9zdAAABcoAAAAAAGQACHJvb3RfcGlkZ2QADW5vbm9kZUBub2hvc3QAAAXKAAAAAABkAAlyb290X3ZpZXdkACVFbGl4aXIuUGhvZW5peC5MaXZlVmlld1Rlc3QuRmxhc2hMaXZlZAAGcm91dGVyZAAiRWxpeGlyLlBob2VuaXguTGl2ZVZpZXdUZXN0LlJvdXRlcmQAB3Nlc3Npb250AAAAAGQABHZpZXdkACpFbGl4aXIuUGhvZW5peC5MaXZlVmlld1Rlc3QuRmxhc2hDaGlsZExpdmVkAAZzaWduZWRuBgDfUA8TcQE.ZuuisoRnufkasnn71DLUmOKY-me2-4XYL0kxuEPgAVw\" data-phx-static=\"\" data-phx-view=\"LiveViewTest.FlashChildLive\" id=\"flash-child\"></div>]\n"
     right: "component[]:info"
     stacktrace:
       test/phoenix_live_view/integrations/flash_test.exs:205: (test)
```
Although I'm not sure that's the correct way of doing that.

Reply:
`clear_flash` should mark the socket as changed and that should be enough to re-render it. What does `Utils.changed?(component_socket)` return?

Reply:
`Utils.changed?(component_socket)` returns `false` and the result of `Diff.with_component` returns:
```elixir
{%{
   c: %{
     0 => %{
       0 => "flash-component",
       1 => "",
       2 => "",
       :s => ["<div id=\"", "\">\ncomponent[", "]:info\ncomponent[",
        "]:error\n</div>\n"]
     }
   }
 },
 {%{
    {Phoenix.LiveViewTest.FlashComponent, "flash-component"} => {0,
     %{
       flash: %{},
       id: "flash-component",
       live_action: nil,
       live_module: Phoenix.LiveViewTest.FlashLive,
       uri: "http://localhost:4000/flash-root"
     }, %{}, {269485866279083608355603410576947787498, %{}}}
  }, %{0 => {Phoenix.LiveViewTest.FlashComponent, "flash-component"}}, 1},
 {nil, %{}}}
```

Reply:
We need to check then why it is not returning as changed. In particular, the flash value before and after calling clear_flash.

Reply:
Forcing the assign for empty flash fixes that:
```elixir
  def assign(%Socket{} = socket, key, value) do
    case socket do
      %{assigns: %{:flash => %{}}} -> force_assign(socket, key, value)
      %{assigns: %{^key => ^value}} -> socket
      %{} -> force_assign(socket, key, value)
    end
```

Reply:
If you have to force, it means it is not really changing. The value before and after is the same. Maybe it is an issue with the test?

Reply:
I believe the test is correct, please see latest commit https://github.com/phoenixframework/phoenix_live_view/pull/724/commits/19829bad2fa126655735fdec9243eea0cd5aafdd

It was returning the unchanged `socket` (first clause) and not marking the socket as changed.

Reply:
I am unease about special casing flash there. I will take a look at it later and see if I can think of alternative approaches. Thanks!

Reply:
No problem, and thanks for your patience and support :)